zipkin-server/src/main/java/zipkin2/server/internal/ZipkinGrpcCollector.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2015-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.server.internal;
+
+import com.linecorp.armeria.common.grpc.protocol.AbstractUnaryGrpcService;
+import com.linecorp.armeria.spring.ArmeriaServerConfigurator;
+import java.util.concurrent.CompletableFuture;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import zipkin2.Callback;
+import zipkin2.codec.SpanBytesDecoder;
+import zipkin2.collector.Collector;
+import zipkin2.collector.CollectorMetrics;
+import zipkin2.collector.CollectorSampler;
+import zipkin2.storage.StorageComponent;
+
+/** Collector for receiving spans on a gRPC endpoint. */
+@ConditionalOnProperty(name = "zipkin.collector.grpc.enabled") // disabled by default
+final class ZipkinGrpcCollector {
```

Comment:
@ewhauser can't get much smaller and still be java, right?

Reply:
👍